spring-integration-core/src/main/java/org/springframework/integration/store/MessageMetadata.java

```diff
@@ -29,10 +29,14 @@
 
 	private static final long serialVersionUID = 1L;
 
-	private final UUID messageId;
+	private UUID messageId;
```

Comment:
See the `private` empty ctor for Jackson deserialization.
Not sure if it will be able to inject value to the `final` property if we assign it to the `null` from that ctor...
I can take a look if that worry you a lot.